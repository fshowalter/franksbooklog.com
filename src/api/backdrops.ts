import { getImage } from "astro:assets";

import { normalizeSources } from "./utils/normalizeSources";

export type BackdropImageProps = {
  src: string;
  srcSet: string;
};

const images = import.meta.glob<{ default: ImageMetadata }>(
  "/content/assets/backdrops/*.png",
);

export async function getBackdropImageProps(
  slug: string,
  {
    height,
    width,
  }: {
    height: number;
    width: number;
  },
): Promise<BackdropImageProps> {
  const backdropFile = await getBackdropFile(slug);

  const optimizedImage = await getImage({
    format: "avif",
    height: height,
    quality: 80,
    src: backdropFile.default,
    width: width,
    widths: [0.3, 0.5, 0.8, 1].map((w) => w * width),
  });

  return {
    src: normalizeSources(optimizedImage.src),
    srcSet: normalizeSources(optimizedImage.srcSet.attribute),
  };
}

async function getBackdropFile(slug: string) {
  let backdropFilePath = Object.keys(images).find((path) => {
    return path.endsWith(`${slug}.png`);
  })!;

  backdropFilePath =
    backdropFilePath ||
    Object.keys(images).find((path) => {
      return path.endsWith(`default.png`);
    })!;

  return await images[backdropFilePath]();
}
