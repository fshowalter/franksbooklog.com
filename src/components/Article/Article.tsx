import { SolidBackdrop } from "src/components/Backdrop";
import { Layout } from "src/components/Layout";
import { LongFormText } from "src/components/LongFormText";
import type { MoreReviewsValue } from "src/components/MoreReviews";
import { MoreReviews } from "src/components/MoreReviews";
import { SubHeading } from "src/components/SubHeading";

export interface Props {
  content: string | null;
  title: string;
  deck: string;
  recentReviews: MoreReviewsValue[];
}

export function Article({
  title,
  content,
  recentReviews,
  deck,
}: Props): JSX.Element {
  return (
    <Layout>
      <article>
        <SolidBackdrop title={title} deck={deck} />
        <section className="flex flex-col items-center pb-32 pt-16">
          <div className="px-container">
            <LongFormText text={content} className="max-w-prose" />
          </div>
        </section>
      </article>
      <div
        data-pagefind-ignore
        className="flex w-full flex-col items-center gap-y-12 bg-subtle pb-32 pt-16 tablet:pt-8 desktop:gap-y-24"
      >
        <MoreReviews values={recentReviews}>
          <SubHeading as="h2">
            Recent{" "}
            <a href="/reviews/" className="text-accent">
              Reviews
            </a>
          </SubHeading>
        </MoreReviews>
      </div>
    </Layout>
  );
}
