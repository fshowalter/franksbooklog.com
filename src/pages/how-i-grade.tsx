import { graphql } from "gatsby";
import { ArticlePage } from "../components/ArticlePage";
import { HeadBuilder } from "../components/HeadBuilder";

export function Head(): JSX.Element {
  return (
    <HeadBuilder
      pageTitle="How I Grade"
      description="The criteria I use to rate movies on this site."
      image={null}
      article
    />
  );
}

export default function HowIGradePage({
  data,
}: {
  data: Queries.HowIGradePageQuery;
}): JSX.Element {
  return (
    <ArticlePage
      image={data.backdrop}
      alt="Empty cinema seats."
      article={data.page}
    />
  );
}

export const pageQuery = graphql`
  query HowIGradePage {
    backdrop: file(absolutePath: { regex: "/backdrops/how-i-grade.png$/" }) {
      ...ArticlePageBackdrop
    }
    page: markdownRemark(frontmatter: { slug: { eq: "how-i-grade" } }) {
      ...ArticlePage
    }
  }
`;
