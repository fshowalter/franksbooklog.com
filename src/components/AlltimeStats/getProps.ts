import { alltimeStats } from "src/api/alltimeStats";
import { getFixedCoverImageProps } from "src/api/covers";
import { allStatYears } from "src/api/yearStats";
import { ListItemCoverImageConfig } from "src/components/ListItemCover";

import { type Props } from "./AlltimeStats";

export async function getProps(): Promise<Props> {
  const stats = await alltimeStats();
  const distinctStatYears = await allStatYears();

  return {
    stats,
    mostReadAuthors: await Promise.all(
      stats.mostReadAuthors.map(async (author) => {
        return {
          ...author,
          readings: await Promise.all(
            author.readings.map(async (reading) => {
              return {
                ...reading,
                coverImageProps: await getFixedCoverImageProps(
                  reading,
                  ListItemCoverImageConfig,
                ),
              };
            }),
          ),
        };
      }),
    ),
    distinctStatYears,
  };
}
