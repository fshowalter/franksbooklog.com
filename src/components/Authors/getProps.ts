import { allAuthorsWithReviews } from "~/api/authors";
import { getAvatarImageProps } from "~/api/avatars";
import { ListItemAvatarImageConfig } from "~/components/ListItemAvatar";

import type { ListItemValue, Props } from "./Authors";

export async function getProps(): Promise<Props> {
  const authors = await allAuthorsWithReviews();

  authors.sort((a, b) => a.sortName.localeCompare(b.sortName));

  const values = await Promise.all(
    authors.map(async (author) => {
      const value: ListItemValue = {
        avatarImageProps: await getAvatarImageProps(
          author.slug,
          ListItemAvatarImageConfig,
        ),
        name: author.name,
        reviewedWorkCount: author.reviewedWorkCount,
        slug: author.slug,
        sortName: author.sortName,
      };

      return value;
    }),
  );

  return {
    deck: '"There is nothing to writing. All you do is sit down at a typewriter and bleed."',
    initialSort: "name-asc",
    values,
  };
}
