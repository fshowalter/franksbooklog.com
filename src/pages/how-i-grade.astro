---
import { getOpenGraphBackdropSrc } from "src/api/backdrops";
import { Article } from "src/components/Article/Article";
import { getProps } from "src/components/Article/getProps";
import Layout from "src/layouts/Layout.astro";

const slug = "how-i-grade";

const props = await getProps({
  slug,
  alt: "Shelves teeming with books.",
});

const openGraphImageSrc = await getOpenGraphBackdropSrc(slug);
---

<Layout
  title={props.title}
  description="The criteria I use to rate books on this site."
  image={openGraphImageSrc}
  article
>
  <Article {...props} />
</Layout>
