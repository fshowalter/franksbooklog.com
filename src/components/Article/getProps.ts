import { getFluidCoverImageProps } from "~/api/covers";
import { getPage } from "~/api/pages";
import { mostRecentReviews } from "~/api/reviews";
import { MoreReviewsImageConfig } from "~/components/MoreReviews";

import type { Props } from "./Article";

export async function getProps({
  deck,
  slug,
}: {
  deck: string;
  slug: string;
}): Promise<Props> {
  const { content, title } = await getPage(slug);
  const recentReviews = await mostRecentReviews(4);

  return {
    content,
    deck,
    recentReviews: await Promise.all(
      recentReviews.map(async (review) => {
        return {
          ...review,
          coverImageProps: await getFluidCoverImageProps(
            review,
            MoreReviewsImageConfig,
          ),
        };
      }),
    ),
    title,
  };
}
