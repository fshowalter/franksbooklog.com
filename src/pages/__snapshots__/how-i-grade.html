<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" sizes="any" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <link rel="manifest" href="/site.webmanifest" />
    <link rel="sitemap" href="/sitemap-index.xml" />
    <meta
      name="description"
      content="The criteria I use to rate books on this site."
    />
    <meta
      name="og:image"
      content="/_image?/content%2Fassets%2Fbackdrops%2Fhow-i-grade.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&#38;w=1200&#38;h=675&#38;q=80&#38;f=jpeg"
    />
    <meta property="og:url" content="https://www.franksbooklog.com//" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="How I Grade" />
    <meta
      property="og:description"
      content="The criteria I use to rate books on this site."
    />
    <title>How I Grade | Frank&#39;s Book Log</title>
  </head>
  <body>
    <div data-pagefind-body>
      <div>
        <a
          href="#content"
          class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
          >Skip to content</a
        >
        <div class="flex min-h-full w-full flex-col bg-default">
          <header
            class="z-20 flex w-full items-center justify-between px-container py-4 tablet:p-6 desktop:inset-x-0 desktop:z-40 desktop:flex-row desktop:flex-wrap desktop:px-16 desktop:py-8 desktop:text-left"
            style="position: absolute"
          >
            <div class="flex flex-col">
              <div
                class="whitespace-nowrap font-normal leading-8"
                style="font-size: 1.5625rem"
              >
                <a href="/">Frank&#x27;s Book Log</a>
              </div>
              <p class="w-full pl-px text-sm italic leading-4 opacity-85">
                Literature is a relative term.
              </p>
            </div>
            <div class="flex items-center">
              <nav class="hidden w-full desktop:block desktop:w-auto">
                <ul class="flex flex-wrap justify-start gap-x-6 text-xl">
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/"
                      >Home</a
                    >
                  </li>
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/how-i-grade/"
                      >How I Grade</a
                    >
                  </li>
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/reviews/"
                      >Reviews</a
                    >
                  </li>
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/readings/"
                      >Reading Log</a
                    >
                  </li>
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/authors/"
                      >Authors</a
                    >
                  </li>
                  <li class="tracking-serif-wide block whitespace-nowrap">
                    <a
                      class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                      href="/shelf/"
                      >Shelf</a
                    >
                  </li>
                </ul>
              </nav>
              <div class="z-[1000]">
                <button
                  data-open-modal="true"
                  disabled=""
                  aria-label="Search"
                  aria-keyshortcuts="Control+K"
                  class="flex items-center justify-between overflow-hidden text-sm leading-6 ring-default desktop:ml-6"
                  type="button"
                  title="Search: Control+K"
                >
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke-width="1.5"
                    stroke="currentColor"
                    class="size-6 desktop:size-7"
                  >
                    <path
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
                    ></path>
                  </svg>
                </button>
              </div>
              <input type="checkbox" id="mobile-nav" class="hidden" /><label
                for="mobile-nav"
                class="hamburger-icon desktop:hidden"
                ><span
                  style="background: #fff"
                  class="hamburger-icon-bars"
                ></span
              ></label>
              <ul
                class="hamburger-menu flex flex-col items-start gap-y-5 text-left text-inverse desktop:hidden"
              >
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/">Home</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/how-i-grade/">How I Grade</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/reviews/">Reviews</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/readings/">Reading Log</a>
                  <ol class="mt-4">
                    <li
                      class="text-inverse-subtle mb-2 font-sans text-xs uppercase tracking-wide"
                    >
                      <a href="/readings/stats/">Stats</a>
                    </li>
                  </ol>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/authors/">Authors</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a href="/shelf/">Shelf</a>
                </li>
              </ul>
            </div>
          </header>
          <main class="grow" id="content">
            <article>
              <header
                class="min-h-[240px] desktop:min-h-[clamp(640px,50vh,1350px)] relative flex w-full flex-col content-start items-center justify-end gap-6 bg-canvas pb-8 pt-40 tablet:pb-10 tablet:pt-40 desktop:pb-16 desktop:pt-40"
              >
                <div
                  class="z-10 mx-auto flex w-full max-w-screen-max flex-col px-container"
                >
                  <h1
                    class="font-sans text-2xl font-bold uppercase tracking-widest desktop:text-7xl"
                  >
                    How I Grade
                  </h1>
                  <p class="mt-1 text-base desktop:my-4 desktop:text-xl">
                    &quot;A work of art is never completed, only
                    abandoned.&quot;
                  </p>
                </div>
              </header>
              <section class="flex flex-col items-center pb-32 pt-16">
                <div class="px-container">
                  <div
                    class="rendered-markdown font-light text-md/7 tracking-0.3px tablet:text-xl/8 max-w-prose"
                  >
                    <p>
                      I apply the following criteria to both long and short form
                      works. It’s a work-in-progress.
                    </p>
                    <ul>
                      <li>
                        <h2>great</h2>
                        <p>
                          <img src="/svg/5-stars.svg" alt="5 stars" /> I loved
                          it. A must-read. Perfect, or almost perfect. These are
                          the works I recommend to everyone. A hard rating to
                          earn.
                        </p>
                      </li>
                      <li>
                        <h2>good</h2>
                        <p>
                          <img src="/svg/4-stars.svg" alt="4 stars" /> I liked
                          it. Unless you hate the genre or author, read it.
                          These are the works I recommend to most folks. They
                          may not resonate like the 5-star selections, but they
                          don’t disappoint.
                        </p>
                      </li>
                      <li>
                        <h2>okay</h2>
                        <p>
                          <img src="/svg/3-stars.svg" alt="3 stars" /> I
                          wouldn’t recommend it, but I wouldn’t recommend
                          <em>against</em> it either. Most works fall into this
                          category.
                        </p>
                      </li>
                      <li>
                        <h2>bad</h2>
                        <p>
                          <img src="/svg/2-stars.svg" alt="2 stars" /> I didn’t
                          like it. For a long-form work, this hints at a
                          disappointing ending. For a shorter work, it
                          underwhelmed. Approach with caution.
                        </p>
                      </li>
                      <li>
                        <h2>awful</h2>
                        <p>
                          <img src="/svg/1-star.svg" alt="1 star" /> I hated it.
                          Another a hard rating to earn. Almost impossible for
                          long works as they’re sooner abandoned.
                        </p>
                      </li>
                    </ul>
                    <hr />
                    <h2>On Abandoning Books</h2>
                    <p>
                      Fifty-three pages into the 320 page novel, I closed it and
                      moved on.
                    </p>
                    <p>It wasn’t easy.</p>
                    <p>
                      I used to finish every book I started. Finishing brought a
                      sense of accomplishment. A comfort in completing a task,
                      checking a box, achieving a goal. Letting go of that
                      illusionary security is hard.
                    </p>
                    <p>But I’m learning.</p>
                    <p>
                      Forcing myself to finish every book didn’t make me a
                      better person. It didn’t make me smarter or more
                      well-rounded. It just made reading a chore rather than a
                      pleasure.
                    </p>
                    <p>
                      Statistics say I’ve got more life behind me than ahead.
                      Which means I’ll never finish all the books I want to
                      read.
                    </p>
                    <p>
                      It’s about opportunity cost. I loved the pulp storytelling
                      of Graham Masterton’s <em>The Devils of D-Day</em>. Mark
                      Richard’s writing in
                      <em>The Ice at the Bottom of the World</em> floored me.
                      Clive Barker’s imagination made
                      <em>The Books of Blood</em> unforgettable. Knowing there
                      are more books out there like those, while I’m forcing
                      myself to trudge through a mediocre read, pushed me to
                      abandon lesser efforts.
                    </p>
                    <p>
                      My shelves hold hundreds of unread books. I’d wager that a
                      quarter of them aren’t worth reading. Maybe more.
                    </p>
                    <p>
                      Some folks give a book a chapter or two. Others give a
                      book a set number of pages. Neither system makes sense to
                      me. I’ve had books start strong then flounder. And how
                      does one apply an arbitrary page count to books of varying
                      lengths? Fifty pages is a quarter of a two hundred page
                      novel, but only five percent of a thousand-page biography.
                    </p>
                    <p>
                      I give a book three hours. The length of a long movie. If
                      a book drags for longer, I’ll abandon it. This doesn’t
                      mean I have to <em>finish</em> it in three hours, just
                      that it can’t go longer than three consecutive hours
                      without holding my interest. Granted, this means I could
                      abandon a thousand-page doorstop with under a hundred
                      pages to go, but I’m okay with that. Failing to recognize
                      a sunk cost is a gambler’s fallacy.
                    </p>
                    <p>
                      Perhaps, as I discover better books, my standards will
                      increase. Today’s excellence might be tomorrow’s schlock.
                      Of course, that would see me abandoning even more books.
                    </p>
                    <p>I can only hope.</p>
                  </div>
                </div>
              </section>
            </article>
            <div
              data-pagefind-ignore="true"
              class="flex w-full flex-col items-center gap-y-12 bg-subtle pb-32 pt-16 tablet:pt-8 desktop:gap-y-24"
            >
              <nav
                data-pagefind-ignore="true"
                class="relative flex w-full flex-col items-center tablet:w-auto"
              >
                <div
                  class="relative mx-auto w-full max-w-screen-max px-container desktop:px-20"
                >
                  <h2
                    class="py-10 font-sans text-xs font-semibold uppercase tracking-wide text-subtle"
                  >
                    Recent<!-- -->
                    <a href="/reviews/" class="text-accent">Reviews</a>
                  </h2>
                  <ul
                    class="flex flex-wrap justify-center gap-x-[3%] gap-y-[6vw] desktop:justify-between"
                  >
                    <li
                      class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                    >
                      <img
                        srcset="
                          /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=44&amp;q=80&amp;f=avif   44w,
                          /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=88&amp;q=80&amp;f=avif   88w,
                          /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;q=80&amp;f=avif 176w,
                          /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=352&amp;q=80&amp;f=avif 352w
                        "
                        src="/_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;h=264&amp;q=80&amp;f=avif"
                        alt=""
                        width="248"
                        height="372"
                        loading="lazy"
                        decoding="async"
                        style="aspect-ratio: 0.66666667"
                      />
                      <div
                        class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                      >
                        <div class="text-center text-md font-medium leading-6">
                          <a
                            href="/reviews/recessional-by-david-mamet/"
                            rel="canonical"
                            class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                            >Recessional</a
                          >
                        </div>
                        <div
                          class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                        >
                          2022<!-- -->
                          |<!-- -->
                          <span class="whitespace-nowrap">Nonfiction</span>
                        </div>
                        <p class="text-center text-base text-subtle">
                          by
                          <!-- -->David Mamet
                        </p>
                        <picture
                          ><source
                            srcset="/svg/2-stars-dark.svg"
                            media="(prefers-color-scheme: dark)" />
                          <img
                            src="/svg/2-stars.svg"
                            alt="D: 2 stars (out of 5)"
                            height="18"
                            width="90"
                            class="mt-2"
                        /></picture>
                      </div>
                    </li>
                    <li
                      class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                    >
                      <img
                        srcset="
                          /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=44&amp;q=80&amp;f=avif   44w,
                          /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=88&amp;q=80&amp;f=avif   88w,
                          /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;q=80&amp;f=avif 176w,
                          /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=352&amp;q=80&amp;f=avif 352w
                        "
                        src="/_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;h=264&amp;q=80&amp;f=avif"
                        alt=""
                        width="248"
                        height="372"
                        loading="lazy"
                        decoding="async"
                        style="aspect-ratio: 0.66666667"
                      />
                      <div
                        class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                      >
                        <div class="text-center text-md font-medium leading-6">
                          <a
                            href="/reviews/fast-forward-by-stephen-morris/"
                            rel="canonical"
                            class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                            >Fast Forward</a
                          >
                        </div>
                        <div
                          class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                        >
                          2020<!-- -->
                          |<!-- -->
                          <span class="whitespace-nowrap">Nonfiction</span>
                        </div>
                        <p class="text-center text-base text-subtle">
                          by
                          <!-- -->Stephen Morris
                        </p>
                        <picture
                          ><source
                            srcset="/svg/4-half-stars-dark.svg"
                            media="(prefers-color-scheme: dark)" />
                          <img
                            src="/svg/4-half-stars.svg"
                            alt="A-: 4.5 stars (out of 5)"
                            height="18"
                            width="90"
                            class="mt-2"
                        /></picture>
                      </div>
                    </li>
                    <li
                      class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                    >
                      <img
                        srcset="
                          /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=44&amp;q=80&amp;f=avif   44w,
                          /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=88&amp;q=80&amp;f=avif   88w,
                          /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;q=80&amp;f=avif 176w,
                          /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=352&amp;q=80&amp;f=avif 352w
                        "
                        src="/_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;h=264&amp;q=80&amp;f=avif"
                        alt=""
                        width="248"
                        height="372"
                        loading="lazy"
                        decoding="async"
                        style="aspect-ratio: 0.66666667"
                      />
                      <div
                        class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                      >
                        <div class="text-center text-md font-medium leading-6">
                          <a
                            href="/reviews/gothic-by-philip-fracassi/"
                            rel="canonical"
                            class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                            >Gothic</a
                          >
                        </div>
                        <div
                          class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                        >
                          2023<!-- -->
                          |<!-- -->
                          <span class="whitespace-nowrap">Novel</span>
                        </div>
                        <p class="text-center text-base text-subtle">
                          by
                          <!-- -->Philip Fracassi
                        </p>
                        <picture
                          ><source
                            srcset="/svg/4-stars-dark.svg"
                            media="(prefers-color-scheme: dark)" />
                          <img
                            src="/svg/4-stars.svg"
                            alt="B+: 4 stars (out of 5)"
                            height="18"
                            width="90"
                            class="mt-2"
                        /></picture>
                      </div>
                    </li>
                    <li
                      class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                    >
                      <img
                        srcset="
                          /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=44&amp;q=80&amp;f=avif   44w,
                          /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=88&amp;q=80&amp;f=avif   88w,
                          /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;q=80&amp;f=avif 176w,
                          /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=352&amp;q=80&amp;f=avif 352w
                        "
                        src="/_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=176&amp;h=264&amp;q=80&amp;f=avif"
                        alt=""
                        width="248"
                        height="372"
                        loading="lazy"
                        decoding="async"
                        style="aspect-ratio: 0.66666667"
                      />
                      <div
                        class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                      >
                        <div class="text-center text-md font-medium leading-6">
                          <a
                            href="/reviews/record-play-pause-by-stephen-morris/"
                            rel="canonical"
                            class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                            >Record Play Pause</a
                          >
                        </div>
                        <div
                          class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                        >
                          2019<!-- -->
                          |<!-- -->
                          <span class="whitespace-nowrap">Nonfiction</span>
                        </div>
                        <p class="text-center text-base text-subtle">
                          by
                          <!-- -->Stephen Morris
                        </p>
                        <picture
                          ><source
                            srcset="/svg/5-stars-dark.svg"
                            media="(prefers-color-scheme: dark)" />
                          <img
                            src="/svg/5-stars.svg"
                            alt="A: 5 stars (out of 5)"
                            height="18"
                            width="90"
                            class="mt-2"
                        /></picture>
                      </div>
                    </li>
                  </ul>
                </div>
              </nav>
            </div>
          </main>
          <footer
            class="bg-footer flex flex-wrap items-start justify-between gap-[10%] px-container py-20 text-inverse tablet:px-12 tablet:pt-10 desktop:p-20"
          >
            <div class="flex flex-col pb-12">
              <div class="flex flex-col">
                <div
                  class="whitespace-nowrap font-normal leading-8"
                  style="font-size: 1.5625rem"
                >
                  <a href="/">Frank&#x27;s Book Log</a>
                </div>
                <p class="w-full pl-px text-sm italic leading-4 opacity-85">
                  Literature is a relative term.
                </p>
              </div>
            </div>
            <div
              class="flex flex-col gap-20 pb-20 pt-10 tablet:basis-1/2 tablet:pt-20"
            >
              <ul class="flex w-full flex-col gap-y-10 text-inverse max:w-auto">
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/">Home</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/how-i-grade/"
                    >How I Grade</a
                  >
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/reviews/">Reviews</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/readings/">Reading Log</a>
                  <ol class="mt-4">
                    <li
                      class="text-inverse-subtle mb-2 font-sans text-xs uppercase tracking-wide"
                    >
                      <a class="hover:text-inverse" href="/readings/stats/"
                        >Stats</a
                      >
                    </li>
                  </ol>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/authors/">Authors</a>
                </li>
                <li class="block w-1/2 whitespace-nowrap text-2xl">
                  <a class="hover:text-accent" href="/shelf/">Shelf</a>
                </li>
              </ul>
              <a
                href="#top"
                class="mb-8 w-full max-w-button bg-canvas py-5 text-center font-sans text-xs uppercase tracking-wide text-default hover:bg-inverse hover:text-inverse"
                >To the top</a
              >
              <p class="text-inverse-subtle w-full font-normal leading-4">
                All covers used in accordance with the<!-- -->
                <a
                  href="http://www.copyright.gov/title17/92chap1.html#107"
                  class="text-inherit underline decoration-2 underline-offset-4 hover:bg-default hover:text-default"
                  >Fair Use Law.</a
                >
              </p>
            </div>
          </footer>
        </div>
      </div>
    </div>
    <dialog
      class="relative m-0 h-screen max-h-none w-screen max-w-unset border border-default bg-subtle shadow-lg open:flex open:flex-col tablet:mx-auto tablet:mt-16 tablet:h-min tablet:max-h-[min(100vh_-_128px,900px)] tablet:min-h-60 tablet:max-w-[min(100vw_-_64px,36rem)] tablet:rounded-xl desktop:min-h-0"
      aria-label="Search"
    >
      <div data-dialog-frame class="contents">
        <button
          data-close-modal
          class="px-4 font-sans text-sm font-light hover:bg-inverse hover:text-inverse focus:outline-accent"
        >
          Cancel
        </button>
        <div class="contents"><div id="pagefind__search"></div></div>
        <div
          class="hidden px-[var(--search-dialog-padding)] py-4 font-sans text-xs font-light tracking-wide text-subtle desktop:block"
        >
          <kbd class="rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >Tab</kbd
          >
          to navigate
          <kbd
            class="ml-4 rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >ESC</kbd
          >
          to close
        </div>
      </div>
    </dialog>
    <script>
      (() => {
        const openBtn = document.querySelector("button[data-open-modal]");
        if (!openBtn) return;
        if (/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)) {
          openBtn.setAttribute("aria-keyshortcuts", "Meta+K");
          openBtn.setAttribute("title", `Search: ⌘K`);
        }
      })();
    </script>
  </body>
</html>
